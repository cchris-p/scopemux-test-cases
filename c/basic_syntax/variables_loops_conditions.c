/**
 * @file variables_loops_conditions.c
 * @brief Demonstration of variables, loops, and conditional statements in C
 *
 * This example shows:
 * - Various variable declarations and types
 * - for, while, and do-while loops
 * - if, else if, else conditions
 * - switch statements
 */

#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>

/**
 * @brief Program entry point
 * @return Exit status code
 */
int main() {
  // Variable declarations
  int i = 0;
  float f = 3.14f;
  double d = 2.71828;
  char c = 'A';
  bool b = true;
  int array[5] = {1, 2, 3, 4, 5};

  // For loop
  printf("For loop example:\n");
  for (i = 0; i < 5; i++) {
    printf("array[%d] = %d\n", i, array[i]);
  }

  // While loop
  printf("\nWhile loop example:\n");
  i = 0;
  while (i < 5) {
    printf("iteration %d\n", i);
    i++;
  }

  // Do-while loop
  printf("\nDo-while loop example:\n");
  i = 0;
  do {
    printf("iteration %d\n", i);
    i++;
  } while (i < 5);

  // If-else statements
  printf("\nIf-else example:\n");
  if (b) {
    printf("b is true\n");
  } else {
    printf("b is false\n");
  }

  // If-else if-else
  printf("\nIf-else if-else example:\n");
  if (i < 3) {
    printf("i is less than 3\n");
  } else if (i == 5) {
    printf("i is equal to 5\n");
  } else {
    printf("i is greater than 5\n");
  }

  // Switch statement
  printf("\nSwitch example:\n");
  switch (c) {
  case 'A':
    printf("c is 'A'\n");
    break;
  case 'B':
    printf("c is 'B'\n");
    break;
  default:
    printf("c is not 'A' or 'B'\n");
    break;
  }

  return 0;
}
