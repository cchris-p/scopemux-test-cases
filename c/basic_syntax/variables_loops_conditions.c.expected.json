{
  "ast": {
    "children": [
      {
        "children": [],
        "content": "/**\n * @file variables_loops_conditions.c\n * @brief Demonstration of variables, loops, and conditional statements in C\n *\n * This example shows:\n * - Various variable declarations and types\n * - for, while, and do-while loops\n * - if, else if, else conditions\n * - switch statements\n */",
        "end_column": 3,
        "end_line": 10,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 1,
        "type": "DOCSTRING"
      },
      {
        "children": [],
        "content": "#include <stdbool.h>",
        "end_column": 20,
        "end_line": 12,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 12,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "#include <stdio.h>",
        "end_column": 18,
        "end_line": 13,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 13,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "#include <stdlib.h>",
        "end_column": 19,
        "end_line": 14,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 14,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "/**\n * @brief Program entry point\n * @return Exit status code\n */",
        "end_column": 3,
        "end_line": 19,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 16,
        "type": "DOCSTRING"
      },
      {
        "children": [],
        "content": "// Variable declarations",
        "end_column": 24,
        "end_line": 21,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 21,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "// For loop",
        "end_column": 11,
        "end_line": 29,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 29,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "// While loop",
        "end_column": 13,
        "end_line": 35,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 35,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "// Do-while loop",
        "end_column": 16,
        "end_line": 43,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 43,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "// If-else statements",
        "end_column": 21,
        "end_line": 51,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 51,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "// If-else if-else",
        "end_column": 18,
        "end_line": 59,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 59,
        "type": "COMMENT"
      },
      {
        "children": [],
        "content": "// Switch statement",
        "end_column": 19,
        "end_line": 69,
        "name": "",
        "qualified_name": "",
        "start_column": 1,
        "start_line": 69,
        "type": "COMMENT"
      },
      {
        "children": [],
        "end_column": 1,
        "end_line": 524,
        "name": "main",
        "qualified_name": "main",
        "start_column": 1,
        "start_line": 139,
        "type": "FUNCTION"
      }
    ],
    "end_column": 1,
    "end_line": 84,
    "name": "variables_loops_conditions.c",
    "qualified_name": "variables_loops_conditions.c",
    "start_column": 1,
    "start_line": 1,
    "type": "ROOT"
  },
  "cst": {
    "children": [
      {
        "children": [],
        "end_column": 20,
        "end_line": 12,
        "start_column": 1,
        "start_line": 12,
        "text": "#include <stdbool.h>",
        "type": "include_directive"
      },
      {
        "children": [],
        "end_column": 18,
        "end_line": 13,
        "start_column": 1,
        "start_line": 13,
        "text": "#include <stdio.h>",
        "type": "include_directive"
      },
      {
        "children": [],
        "end_column": 19,
        "end_line": 14,
        "start_column": 1,
        "start_line": 14,
        "text": "#include <stdlib.h>",
        "type": "include_directive"
      },
      {
        "children": [
          {
            "children": [],
            "text": "// Variable declarations",
            "type": "statement"
          },
          {
            "children": [],
            "text": "int i = 0;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "float f = 3.14f;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "double d = 2.71828;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "char c = 'A';",
            "type": "statement"
          },
          {
            "children": [],
            "text": "bool b = true;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "int array[5] = {1, 2, 3, 4, 5};",
            "type": "statement"
          },
          {
            "children": [],
            "text": "// For loop",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"For loop example:\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "for (i = 0; i < 5; i++) {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"array[%d] = %d\\n\", i, array[i]);",
            "type": "statement"
          },
          {
            "children": [],
            "text": "}",
            "type": "statement"
          },
          {
            "children": [],
            "text": "// While loop",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"\\nWhile loop example:\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "i = 0;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "while (i < 5) {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"iteration %d\\n\", i);",
            "type": "statement"
          },
          {
            "children": [],
            "text": "i++;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "}",
            "type": "statement"
          },
          {
            "children": [],
            "text": "// Do-while loop",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"\\nDo-while loop example:\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "i = 0;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "do {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"iteration %d\\n\", i);",
            "type": "statement"
          },
          {
            "children": [],
            "text": "i++;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "} while (i < 5);",
            "type": "statement"
          },
          {
            "children": [],
            "text": "// If-else statements",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"\\nIf-else example:\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "if (b) {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"b is true\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "} else {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"b is false\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "}",
            "type": "statement"
          },
          {
            "children": [],
            "text": "// If-else if-else",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"\\nIf-else if-else example:\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "if (i < 3) {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"i is less than 3\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "} else if (i == 5) {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"i is equal to 5\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "} else {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"i is greater than 5\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "}",
            "type": "statement"
          },
          {
            "children": [],
            "text": "// Switch statement",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"\\nSwitch example:\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "switch (c) {",
            "type": "statement"
          },
          {
            "children": [],
            "text": "case 'A':",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"c is 'A'\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "break;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "case 'B':",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"c is 'B'\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "break;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "default:",
            "type": "statement"
          },
          {
            "children": [],
            "text": "printf(\"c is not 'A' or 'B'\\n\");",
            "type": "statement"
          },
          {
            "children": [],
            "text": "break;",
            "type": "statement"
          },
          {
            "children": [],
            "text": "}",
            "type": "statement"
          },
          {
            "children": [],
            "text": "return 0;",
            "type": "statement"
          }
        ],
        "end_column": 1,
        "end_line": 84,
        "name": "main",
        "start_column": 1,
        "start_line": 20,
        "text": "int main() {\n  // Variable declarations\n  int i = 0;\n  float f = 3.14f;\n  double d = 2.71828;\n  char c = 'A';\n  bool b = true;\n  int array[5] = {1, 2, 3, 4, 5};\n\n  // For loop\n  printf(\"For loop example:\\n\");\n  for (i = 0; i < 5; i++) {\n    printf(\"array[%d] = %d\\n\", i, array[i]);\n  }\n\n  // While loop\n  printf(\"\\nWhile loop example:\\n\");\n  i = 0;\n  while (i < 5) {\n    printf(\"iteration %d\\n\", i);\n    i++;\n  }\n\n  // Do-while loop\n  printf(\"\\nDo-while loop example:\\n\");\n  i = 0;\n  do {\n    printf(\"iteration %d\\n\", i);\n    i++;\n  } while (i < 5);\n\n  // If-else statements\n  printf(\"\\nIf-else example:\\n\");\n  if (b) {\n    printf(\"b is true\\n\");\n  } else {\n    printf(\"b is false\\n\");\n  }\n\n  // If-else if-else\n  printf(\"\\nIf-else if-else example:\\n\");\n  if (i < 3) {\n    printf(\"i is less than 3\\n\");\n  } else if (i == 5) {\n    printf(\"i is equal to 5\\n\");\n  } else {\n    printf(\"i is greater than 5\\n\");\n  }\n\n  // Switch statement\n  printf(\"\\nSwitch example:\\n\");\n  switch (c) {\n  case 'A':\n    printf(\"c is 'A'\\n\");\n    break;\n  case 'B':\n    printf(\"c is 'B'\\n\");\n    break;\n  default:\n    printf(\"c is not 'A' or 'B'\\n\");\n    break;\n  }\n\n  return 0;\n}",
        "type": "function_definition"
      },
      {
        "children": [],
        "end_column": 3,
        "end_line": 10,
        "start_column": 1,
        "start_line": 1,
        "text": "/**\n * @file variables_loops_conditions.c\n * @brief Demonstration of variables, loops, and conditional statements in C\n *\n * This example shows:\n * - Various variable declarations and types\n * - for, while, and do-while loops\n * - if, else if, else conditions\n * - switch statements\n */",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 3,
        "end_line": 19,
        "start_column": 1,
        "start_line": 16,
        "text": "/**\n * @brief Program entry point\n * @return Exit status code\n */",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 24,
        "end_line": 21,
        "start_column": 1,
        "start_line": 21,
        "text": "// Variable declarations",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 11,
        "end_line": 29,
        "start_column": 1,
        "start_line": 29,
        "text": "// For loop",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 13,
        "end_line": 35,
        "start_column": 1,
        "start_line": 35,
        "text": "// While loop",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 16,
        "end_line": 43,
        "start_column": 1,
        "start_line": 43,
        "text": "// Do-while loop",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 21,
        "end_line": 51,
        "start_column": 1,
        "start_line": 51,
        "text": "// If-else statements",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 18,
        "end_line": 59,
        "start_column": 1,
        "start_line": 59,
        "text": "// If-else if-else",
        "type": "comment"
      },
      {
        "children": [],
        "end_column": 19,
        "end_line": 69,
        "start_column": 1,
        "start_line": 69,
        "text": "// Switch statement",
        "type": "comment"
      }
    ],
    "end_column": 1,
    "end_line": 84,
    "start_column": 1,
    "start_line": 1,
    "type": "translation_unit"
  },
  "language": "C"
}